frontend/src/component/project/Project/Import/configure/ConfigurationStage.tsx

```diff
@@ -38,129 +37,133 @@ const MaxSizeMessage = styled(Typography)(({ theme }) => ({
+        <ConditionallyRender
+            condition={activeTab === 'file'}
+            show={
+                <StyledFileDropZone
+                    onSuccess={data => {
+                        setImportPayload(data);
+                        setActiveTab('code');
+                        setToastData({
+                            type: 'success',
+                            title: 'File uploaded',
+                        });
+                    }}
+                    onError={error => {
+                        setImportPayload('');
+                        setToastData({
+                            type: 'error',
+                            title: error,
+                        });
+                    }}
+                    onDragStatusChange={setDragActive}
                 >
-                    Validate
-                </Button>
-                <Button
-                    sx={{ position: 'static', ml: 2 }}
+                    <PulsingAvatar active={dragActive}>
+                        <ArrowUpward fontSize="large" />
+                    </PulsingAvatar>
+                    <DropMessage>
+                        {dragActive
+                            ? 'Drop your file to upload'
+                            : 'Drop your file here'}
+                    </DropMessage>
+                    <SelectFileMessage>
+                        or select a file from your device
+                    </SelectFileMessage>
+                    <Button variant="outlined">Select file</Button>
+                    <MaxSizeMessage>JSON format: max 500 kB</MaxSizeMessage>
+                </StyledFileDropZone>
+            }
+            elseShow={
+                <StyledTextField
+                    label="Exported toggles"
                     variant="outlined"
-                    type="submit"
-                    onClick={onClose}
-                >
-                    Cancel import
-                </Button>
-            </ActionsContainer>
+                    onChange={event => setImportPayload(event.target.value)}
+                    value={importPayload}
+                    multiline
+                    minRows={13}
+                    maxRows={13}
+                />
+            }
+        />
+    );
+};
+
+export const Actions: FC<{
+    onSubmit: () => void;
+    onClose: () => void;
+    disabled: boolean;
+}> = ({ onSubmit, onClose, disabled }) => (
+    <ActionsContainer>
+        <Button
+            sx={{ position: 'static' }}
+            variant="contained"
+            type="submit"
+            onClick={onSubmit}
+            disabled={disabled}
+        >
+            Validate
+        </Button>
+        <Button
+            sx={{ position: 'static', ml: 2 }}
+            variant="outlined"
+            type="submit"
+            onClick={onClose}
+        >
+            Cancel import
+        </Button>
+    </ActionsContainer>
+);
+
+export const ConfigurationStage: FC<{
+    tabs: ReactNode;
```

Comment:
this is a typical signature for "slots not props" technique. ReactNode instead of lots of string fields